packages/main/src/queue/index.ts

```diff
@@ -246,6 +246,72 @@ const handlers = {
         sync(accountId, connectionManager)
       )
     }
+  }),
+
+  flag: handler({
```

Comment:
The `flag` and `unflag` handlers are nearly identical. They could be combined into one `setFlagged` handler with a boolean parameter.